syntax = "proto3";

package aa.proto;

import "frame.proto";
import "polygon.proto";

option cc_enable_arenas = true;

// Processing request
message ProcessFrameRequest {
  Frame frame = 1;
  repeated Polygon polygons = 2;  // List of polygons for detection zones
}

// Processing response
message ProcessFrameResponse {
  Frame result = 1;
}

// Health check
message CheckHealthRequest {}

message CheckHealthResponse {
  bool healthy = 1;
  string status = 2;
}

// Detection service
service DetectorService {
  rpc ProcessFrame(ProcessFrameRequest) returns (ProcessFrameResponse);
  rpc CheckHealth(CheckHealthRequest) returns (CheckHealthResponse);
}
